buildSrc/src/main/java/io/micronaut/guides/feature/DockerPushFeature.java

```diff
@@ -0,0 +1,35 @@
+package io.micronaut.guides.feature;
+
+import io.micronaut.guides.feature.template.dockerPush;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.gradle.GradlePlugin;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.template.RockerWritable;
+
+public abstract class DockerPushFeature implements Feature {
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        if (generatorContext.getBuildTool().isGradle()) {
+            generatorContext.addBuildPlugin(GradlePlugin
+                    .builder()
+                    .id("idea") // ID is considered required. Adding a not harmful plugin
+                    .extension(new RockerWritable(dockerPush.template(getImage())))
+                    .build());
+        }
+    }
```

Comment:
Is this necessary to be able to add 

```groovy
dockerBuild {
    images = ["gcr.io/micronaut-guides/micronautguide:latest"]
}
```

to the build?

Reply:
the `idea` part is a `dirty` workaround. `id` is required.  We have to change `starter` so that `id` is not a required. Or probably better make Micronaut library and Micronaut Application into real features so that we can extend them in guides.